packages/circuit-ui/components/SidePanel/SidePanel.tsx

```diff
@@ -15,147 +15,171 @@
-     */
-    isBottomPanelClosing: boolean;
-    /**
-     * Boolean indicating whether the side panel should be opened without animation.
-     */
-    isInstantOpen: boolean;
-    /**
-     * Boolean indicating whether the side panel should be in desktop or mobile mode.
-     */
-    isMobile: boolean;
-    /**
-     * Boolean indicating whether the side panel is stacked over another panel.
-     */
-    isStacked: boolean;
-    /**
-     * Callback function that is called when the side panel is closed.
-     */
-    onBack?: OnBack;
-    /**
-     * Callback function that is called when the side panel is closed.
-     */
-    onClose: OnClose;
-  };
+export type SidePanelProps = Omit<DialogProps, 'children'> & {
+  /**
+   * The headline of the side panel.
+   */
+  headline: string;
+  /**
+   * Text label for the back button for screen readers.
+   * Important for accessibility.
+   */
+  backButtonLabel?: string;
+  /**
+   * The side panel content. Use a render function when you need access to the
+   * `onBack` and `onClose` functions to close the side panel programmatically.
+   */
+  children: ReactNode | ((props: ChildrenRenderProps) => ReactNode);
+  /**
+   * Boolean indicating whether the side panel should be in desktop or mobile mode.
+   */
+  isMobile: boolean;
+  /**
+   * Callback function that is called when the stacked side panel is closed.
+   */
+  onBack?: OnBack;
+  /**
+   * Callback function that is called when the side panel is closed.
+   */
+  onClose?: OnClose;
+};
 
-export const SidePanel = ({
-  backButtonLabel,
-  children,
-  closeButtonLabel,
-  headline,
-  isBottomPanelClosing,
-  isInstantOpen,
-  isMobile,
-  isStacked,
-  onBack,
-  onClose,
-  ...props
-}: SidePanelProps) => {
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    process.env.NODE_ENV !== 'test'
-  ) {
-    if (!isSufficientlyLabelled(closeButtonLabel)) {
-      throw new AccessibilityError(
-        'SidePanel',
-        'The `closeButtonLabel` prop is missing or invalid.',
+export const SidePanel = forwardRef<HTMLDialogElement, SidePanelProps>(
+  (props, ref) => {
+    const {
+      open,
+      backButtonLabel,
+      children,
+      closeButtonLabel,
+      headline,
+      isMobile,
+      onBack,
+      onCloseEnd,
+      onClose,
+      preventOutsideClickClose,
+      animationDuration = 0,
+      preventEscapeKeyClose,
+      className,
+      ...rest
+    } = useI18n(props, translations);
+    {
+      const [animationClass, setAnimationClass] = useState<string | undefined>(
+        classes.closed,
       );
-    }
-    if (onBack && !isSufficientlyLabelled(backButtonLabel)) {
-      throw new AccessibilityError(
-        'SidePanel',
-        'The `backButtonLabel` prop is missing or invalid.',
+      const [isHeaderSticky, setHeaderSticky] = useState(false);
+      const headerAriaId = useId();
+
+      const [, setAnimating] = useAnimation();
+
+      const triggerAnimation = useCallback(
```

Comment:
I think splitting this into two functions would make the code more readable: one for opening and one for closing the panel. Instead of passing the `isOpening` parameter, you could call the dedicated function and avoid the `if`/`else` branches inside.